src/App.tsx

```diff
@@ -14,11 +14,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+export class App extends React.Component<{}, { selectedGoods: string[] }> {
+  state = {
+    selectedGoods: [],
+  };
+
+  clearAll = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  };
+
+  addItem = (item: string) => {
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods, item],
+    }));
+  };
+
+  removeItem = (item: string) => {
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods.filter(element => element !== item)],
```

Comment:
```suggestion
      selectedGoods: state.selectedGoods.filter(element => element !== item),
```
filter и так делает копию, так что не обязательно делать деструктуризацию